torch/distributed/distributed_c10d.py

```diff
@@ -1225,10 +1238,8 @@ def _new_process_group_helper(
             "created, please use a different group name"
         )
 
-    if not isinstance(timeout, timedelta):
-        raise TypeError(
-            "Expected timeout argument to be of type datetime.timedelta"
-        )
+    # Note: _new_process_group_helper is only called from init_process_group, which always provides a timeout value
+    _check_valid_timeout(timeout)
```

Comment:
isn't the `timeout` argument passed in already checked in `init_process_group`? My thought is we don't need this check in all private APIs (i.e. `_new_process_group_helper` and `_new_group_with_tag`) because they must have been checked in the public API before being passed in.

Reply:
init_process_group checks the timeout and then calls _new_process_group_helper

other 'new_group' apis also call _new_process_Group_helper directly.

My first thought was to move all the checking inside the helper and take it out of all the outer functions.  I still think this is a good idea.  However, this is a larger change since init_process_group also has handling for `pg_options` timeout value.  Should I fix this now or in a separate pr...?